src/solve.jl

```diff
@@ -1,31 +1,39 @@
-abstract type AbstractOptimizationSolution end #experimental; comments welcome
-mutable struct OptimizationSolution{O, Tx, Tf, Tls, Tsb} <: AbstractOptimizationSolution
-    method::O
-    initial_x::Tx
-    minimizer::Tx
+abstract type AbstractOptimizationSolution{T, N} <: AbstractNoTimeSolution{T, N} end
+
+struct OptimizationSolution{T, N, uType, P, A, Tf} <: AbstractOptimizationSolution{T, N}
+    u::uType # minimizer
+    prob::P # optimization problem
+    alg::A # algorithm
     minimum::Tf
-    iterations::Int
-    iteration_converged::Bool
-    ls_success::Tls
-    time_run::Float64
-    stopped_by::Tsb
+    initial_x::Array{Float64,1}
+    retcode::Symbol
+    original::String # original output of the optimizer
 end
 
-function Base.show(io::IO, r::AbstractOptimizationSolution)
-    take = Iterators.take
-    failure_string = "failure"
-    if isa(r.ls_success, Bool) && !r.ls_success
-        failure_string *= " (line search failed)"
-    end
+function build_solution(prob::AbstractNonlinearProblem,
```

Comment:
because this is not a PR to SciMLBase, you're just shadowing and this should be `SciMLBase.build_solution`. Why is this defined here?

Reply:
I forgot to upload the changed GalacticOptim.jl file where the build_solution is from SciMLBase. I edited it in two different places.